src/main/kotlin/de/amplimind/codingchallenge/service/UserService.kt

```diff
@@ -46,6 +46,15 @@ class UserService(
         } ?: throw ResourceNotFoundException("User with email $email was not found")
     }
 
+    /**
+     * Deletes a user by its email
+     * @param email the email of the user to delete
+     */
+    fun deleteUserByEmail(email: String) {
+        val user = this.userRepository.findByEmail(email)
+        user?.let { this.userRepository.delete(it) }
```

Comment:
Wirf einfach eine ResourceNotFoundException wie in der anderen delete methode im controller